test/integration/kubernetes_deploy_test.rb

```diff
@@ -194,6 +194,15 @@ def test_refuses_deploy_to_protected_namespace_with_override_if_pruning_enabled
     @namespace = generated_ns
   end
 
+  def test_deploy_succeeds_with_specific_protected_namespaces
+    generated_ns = @namespace
+    @namespace = 'default' # this should fail if we use the default options for protected namespaces
+    protected_namespaces = %w(kube-system kube-public)
+    assert_deploy_success(deploy_fixtures("hello-cloud", prune: true, protected_namespaces: protected_namespaces))
+  ensure
+    @namespace = generated_ns
```

Comment:
This ensure should clean up the changes made to the 'default' namespace. @namespace is set per test so no need to set it back.

Reply:
There are 3 more tests doing that here; I was assuming this is a safeguard to help prevent setup leak when we run these tests in parallel. I'll check.

Reply:
Hmmm, guess I'm wrong. Though I suspect you should still clean-up default here